src/library/scala/math/Ordering.scala

```diff
@@ -369,7 +370,36 @@ object Ordering extends LowPriorityOrderingImplicits {
 
   /** `Ordering`s for `Float`s.
     *
-    * @define floatOrdering Because the behaviour of `Float`s specified by IEEE is
+    * The behavior of the comparison operations provided by default (implicit)
+    * ordering on `Float` has changed in 2.10.0 and 2.13.0.
+    * Prior to Scala 2.10.0, the `Ordering` instance used the semantics
+    * consistent with `java.lang.Float.compare`.
+    *
+    * Scala 2.10.0 changed the implementation of `lt`, `equiv`, `min` etc to be
+    * IEEE 754 compliant, while keeping `compare` method NOT compliant,
+    * creating an internally inconsistent instance. IEEE 754 specified that
+    * `0.0F == -0.0F`. In addition, all comparisons with `Float.NaN` must return
+    * `false` thus `0.0F < Float.NaN`, `0.0F > Float.NaN`, and
+    * `Float.NaN == Float.NaN` all yield `false`, analogous to `None`.
+    *
+    * Recognizing the limitation of the IEEE 754 semantics in terms of ordering,
+    * Scala 2.13.0 created two instances `Ordering.Float.IeeeOrdering` and
+    * `Ordering.Float.TotalOrdering`, which brings back the `java.lang.Float.compare`
```

Comment:
this wording is a bit ambiguous and kind of makes it seem like both `IeeeOrdering` and `TotalOrdering` both have `Float.compare` semantics, but only the latter does. While it should be obvious that `IeeeOrdering` has IEEE 754 semantics, it might be worth changing the wording to something like

> Scala 2.13.0 created two instances: `Ordering.Float.IeeeOrdering`, which retains the IEEE 754 semantics from Scala 2.12.x, and `Ordering.Float.TotalOrdering`, which brings back the `java.lang.Float.compare` semantics for all operations.

Reply:
👍

Reply:
Addressed in acf7044